### ISSUE TITLE
F601: Repeated tuple keys with different values are not detected

### ISSUE THREAD

**Issue by MatthewHoadSWO** [astral-sh/ruff#5691]
Duplicate tuple keys with different values in a dict are not detected. Not sure if `F601` is the correct rule to file this under since I'm unsure if tuples are technically literals. If not, then it would be really handy to have a rule for this. Flake8 seems to think that duplicate tuple keys fall under F601 though. Reproduced with ruff `0.0.269`, `0.0.273`, `0.0.275`, and `0.0.277`. 

To reproduce: 
Default ruff settings (no `pyproject.toml`)

```sh
>>> ruff --version
ruff 0.0.277
```

`f601_sample.py` 👇 
```python
my_dict = {
    ('a', 'b'): 'asdf',
    ('a', 'b'): 'qwer',
    'a': 'asdf',
    'a': 'qwer',
}
```

Ruff correctly detects the duplicated string literal key, but not the tuple key:
```sh
>>> ruff --isolated f601_sample.py
f601_sample.py:5:5: F601 Dictionary key literal `'a'` repeated
Found 1 error.
```

Flake8 detects both string and tuple key duplication:
```sh
>>> flake8 f601_sample.py 
f601_sample.py:2:5: F601 dictionary key ('a', 'b') repeated with different values
f601_sample.py:3:5: F601 dictionary key ('a', 'b') repeated with different values
f601_sample.py:4:5: F601 dictionary key 'a' repeated with different values
f601_sample.py:5:5: F601 dictionary key 'a' repeated with different values
```

**Comment by dhruvmanila** [comment]
Yeah, this should be supported. For anyone interested, a new `DictionaryKey` type would be added [here](https://github.com/astral-sh/ruff/blob/main/crates/ruff/src/rules/pyflakes/rules/repeated_keys.rs#L129) and the `match` expressions using this value would need to account for it [here](https://github.com/astral-sh/ruff/blob/main/crates/ruff/src/rules/pyflakes/rules/repeated_keys.rs#L135) and [here](https://github.com/astral-sh/ruff/blob/main/crates/ruff/src/rules/pyflakes/rules/repeated_keys.rs#L157).

**Comment by charliermarsh** [comment]
I wonder why we have a dedicated type there rather than just using ComparableExpr.

**Comment by dhruvmanila** [comment]
I haven't looked into `ComparableExpr` much but I think as dictionary keys needs to be hashable we can't consider all of `ComparableExpr` variants. Or, are you referring to something else?

**Comment by charliermarsh** [comment]
I was suggesting just using ComparableExpr and removing DictionaryKey. If a key uses an unhashable Python type, like a set, that’s a problem regardless, right? But it’s not a big deal, it just seems simpler and would also catch things like attribute accesses and subscript accesses.
